src/pybind/mgr/dashboard/controllers/multi_cluster.py

```diff
@@ -153,13 +167,19 @@ def reconnect_cluster(self, url: str, username=None, password=None, token=None,
                 'username': username,
                 'password': password
             }
-            content = self._proxy('POST', url, 'api/auth', payload=payload,
+            try:
+                content = self._proxy('POST', url, 'api/auth', payload=payload,
                                   verify=ssl_verify, cert=ssl_certificate)
-            if 'token' not in content:
-                raise DashboardException(
-                    "Could not authenticate to remote cluster",
-                    http_status_code=400,
-                    component='dashboard')
+                if 'token' not in content:
+                    return content['detail']
+                user_content = self._proxy('GET', url, f'api/user/{username}',
+                                        token=content['token'])
+                if 'status' in user_content and user_content['status'] == '403 Forbidden':
+                    return 'User is not an administrator'
+            except Exception as e:  # pylint: disable=broad-except
+                if '[Errno 111] Connection refused' in str(e):
+                    return 'Connection refused'
+                return 'Connection failed'
```

Comment:
don't return errors. raise Exception whenever there is an error and properly catch it in the UI